text/0000-gatsby-plugin-cli.md

```diff
@@ -0,0 +1,161 @@
+Thanks everyone for checking this out. I've started on the code and am a happy to be the one to implement most/all of this. Thanks for the feedback all keep up the great work on gatsby.
+
+
+- Start Date: 2018-09-28
+- RFC PR: (leave this empty)
+- Gatsby Issue: (leave this empty)
+
+# Summary
+
+I find it  tedious to have to do the same 3 steps every time I add a plugin to gatsby.
+
+* `npm instal *plugin*`
+* Check plugin docs for correct -`gatsby-config.js` syntax
+* modify `gatsby-config.js`
+
+I'm suggesting we create an additional command to add and remove plugins using the `gatsby-cli`.
+
+# Basic example
+
+A new work flow for installing a plugin could look something like this:
+
+* `gatsby plugin add gatsby-transformer-x` (this would run npm install for the plugin and inject an initial config into `gatsby-config.js`)
+
+* For plugins that don't require additional config we're done. If they do I at least have the basics that I can probably understand how to modify from a quick glance. ie. `path: /your/path/to/xyz.whatever` I know what this needs to point to and I can do that without checking docs.
+
+* if needed then the dev can check docs.
+
+
+A new work flow for removing a plugin could look like:
+ 
+ * `gatsby plugin remove gatsby-transformer-x` (This would just remove the config for that plugin and remove the package using npm from the `package.json` and node_modules.)
```

Comment:
It would be nice to have a prompt before executing the actions, something like.

```
You are about to remove gatsby-transformer-x. This will remove it from 'gatsby-config.js' and package.json
Are you sure you want to do this?[Yn]:
```
and to skip this confirmation, for example when running the script on ci, we can have the `-y`.

Reply:
@Khaledgarbaya I like the idea. Especially if we're installing dependencies it'd be a good time to tell the dev that those are also being installed. 

If we're not handing dependencies...I'm struggling to see the benefit.  Mostly in the context that it mimics behavior of apt-get not npm/yarn which gatsby devs are going to be more familiar with using. What specific problem(s) do you see this solving? Thanks

Reply:
Hey @moonmeister,
It's always a good thing to have a confirmation whenever there is deletion because of fat fingers 🙈, especially when you relay on terminal autocomplete so sometimes you might pick the wrong command

Reply:
Excellent point.

Reply:
A confirmation seems like a good idea to me. There's no harm in doing it, and it'll make it very clear how the CLI is changing the project — less magic is a good think, in my opinion.

If we want to get really wild, we could look at settings for the CLI to allow this to be toggled on or off, but that's a discussion for another day.

Reply:
Agreed